src/shaders/MaterialX/color2.h

```diff
@@ -15,340 +15,287 @@ struct color2
+                    );
+
 }
 
 color2 exponent(color2 in, float amount)
 {
-    color2 out = in;
-    out.r = exponent(in.r, amount);
-    out.g = exponent(in.g, amount);
-    return out;
+    return color2 (exponent(in.r, amount),
+                    exponent(in.g, amount)
+                    );
 }
 
 color2 max(color2 in, color2 amount)
 {
-    color2 out = in;
-    out.r = max(in.r, amount.r);
-    out.g = max(in.g, amount.g);
-    return out;
+    return color2 (max(in.r, amount.r),
+                    max(in.g, amount.g)
+                    );
 }
 
 color2 max(color2 in, float amount)
 {
-    color2 out = in;
-    out.r = max(in.r, amount);
-    out.g = max(in.g, amount);
-    return out;
+    return color2 (max(in.r, amount),
+                    max(in.g, amount)
+                    );
 }
 
-color2 multiply(color2 in1, color2 amount)
+color2 multiply(color2 in, color2 amount)
 {
-    color2 out = in1;
-    out.r = in1.r * amount.r;
-    out.g = in1.g * amount.g;
-    return out;
+    return color2 (in.r * amount.r,
+                    in.g * amount.g
+                    );
 }
 
-color2 multiply(color2 in1, float amount)
+color2 multiply(color2 in, float amount)
 {
-    color2 out = in1;
-    out.r = in1.r * amount;
-    out.g = in1.g * amount;
-    return out;
+    return color2 (in.r * amount,
+                    in.g * amount
+                    );
 }
 
 color2 min(color2 in, color2 amount)
 {
-    color2 out = in;
-    out.r = min(in.r, amount.r);
-    out.g = min(in.g, amount.g);
-    return out;
+    return color2 (min(in.r, amount.r),
+                    min(in.g, amount.g)
+                    );
 }
 
 color2 min(color2 in, float amount)
 {
-    color2 out = in;
-    out.r = min(in.r, amount);
-    out.g = min(in.g, amount);
-    return out;
+    return color2 (min(in.r, amount),
+                    min(in.g, amount)
+                    );
 }
 
 color2 fmod(color2 in, color2 amount)
 {
-    color2 out = in;
-    out.r = fmod(in.r, amount.r);
-    out.g = fmod(in.g, amount.g);
-    return out;
+    return color2 (fmod(in.r, amount.r),
+                    fmod(in.g, amount.g)
+                    );
 }
 
 color2 fmod(color2 in, float amount)
 {
-    color2 out = in;
-    out.r = fmod(in.r, amount);
-    out.g = fmod(in.g, amount);
-    return out;
+    return color2 (fmod(in.r, amount),
+                    fmod(in.g, amount)
+                    );
 }
 
 color2 unpremult(color2 in)
 {
-    color2 out = in;
-    out.r = in.r / in.g;
-    return out;
+    return color2 (in.r / in.g, in.g);
 }
 
 color2 premult(color2 in)
 {
-    color2 out = in;
-    out.r = in.r * in.g;
-    return out;
+    return color2 (in.r * in.g, in.g);
 }
 
 color2 cout(color2 fg, color2 bg)
 {
-    color2 out;
-    out.r = fg.r * (1 - bg.g);
-    out.g = fg.g * (1 - bg.g);
-    return out;
+    return color2 (fg.r * (1 - bg.g),
+                   fg.g * (1 - bg.g)
+                   );
 }
 
 color2 over(color2 fg, color2 bg)
```

Comment:
Is this used? It sure is weird. For a color2 (where you have called the components r and g) does it make sense to magically treat g as if it were alpha?